src/WebJobs.Script.WebHost/Middleware/SystemTraceMiddleware.cs

```diff
@@ -27,58 +27,83 @@ public SystemTraceMiddleware(RequestDelegate next, ILogger<SystemTraceMiddleware
 
         public async Task Invoke(HttpContext context)
         {
-            SetRequestId(context.Request);
+            var requestId = SetRequestId(context.Request);
 
-            var sw = new Stopwatch();
-            sw.Start();
-            var request = context.Request;
-            var details = new JObject
-            {
-                { "requestId", request.GetRequestId() },
-                { "method", request.Method.ToString() },
-                { "uri", request.Path.ToString() }
-            };
+            var sw = Stopwatch.StartNew();
+            string method = context.Request.Method;
+            string path = context.Request.Path.ToString();
+            string userAgent = context.Request.GetHeaderValueOrDefault("User-Agent");
             var logData = new Dictionary<string, object>
             {
-                [ScriptConstants.LogPropertyActivityIdKey] = request.GetRequestId()
+                [ScriptConstants.LogPropertyActivityIdKey] = requestId
             };
-            _logger.Log(LogLevel.Information, 0, logData, null, (s, e) => $"Executing HTTP request: {details}");
+            _logger.Log(LogLevel.Information, 0, logData, null, (s, e) =>
+            {
```

Comment:
Here's an example of an expensive log statement, that is now being moved off of the request hotpath. Important when doing this to consider what is being captured by the closure, to avoid holding on to objects longer than necessary (e.g the request).

Reply:
No, the delegate is no longer called on the hotpath. Logging is now deferred and flushed in the background.

Reply:
I've moved both of these to LogMessage.Defines, with a json format template. Will be in next iteration